plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTransactionManager.java

```diff
@@ -53,6 +53,11 @@ public void commit(ConnectorTransactionHandle transaction)
     {
         MemoizedMetadata deltaLakeMetadata = transactions.remove(transaction);
         checkArgument(deltaLakeMetadata != null, "no such transaction: %s", transaction);
+        deltaLakeMetadata.optionalGet().ifPresent(metadata -> {
+            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+                metadata.commit();
```

Comment:
*unrelated question:* Why is `ThreadContextClassLoader` required for `metadata.commit()`?

Reply:
probably to ensure that metadata.commit() call will be executed in the context of the delta lake plugin classloader